src/screenshots/capturer.js

```diff
@@ -1,66 +1,112 @@
 import { join as joinPath, dirname } from 'path';
 import promisify from '../utils/promisify';
+import sanitize from 'sanitize-filename';
 import mkdirp from 'mkdirp';
+import { generateThumbnail } from 'testcafe-browser-natives';
 
 var ensureDir = promisify(mkdirp);
 
 const PNG_EXTENSION_RE = /(\.png)$/;
 
 
 export default class Capturer {
-    constructor (baseScreenshotsPath, testScreenshotsPath, testEntry, connection) {
-        this.enabled             = !!baseScreenshotsPath;
-        this.baseScreenshotsPath = baseScreenshotsPath;
-        this.testScreenshotsPath = testScreenshotsPath;
-        this.testEntry           = testEntry;
-        this.provider            = connection.provider;
-        this.id                  = connection.id;
+    constructor (baseScreenshotsPath, testEntry, connection, namingOptions) {
+        this.enabled              = !!baseScreenshotsPath;
+        this.baseScreenshotsPath  = baseScreenshotsPath;
+        this.testEntry            = testEntry;
+        this.provider             = connection.provider;
+        this.id                   = connection.id;
+        this.startDate            = namingOptions.startDate;
+        this.userAgentName        = namingOptions.userAgentName;
+        this.testDirName          = Capturer._getTestDirName(namingOptions.testIndex, namingOptions.quarantineAttempt);
+        this.screenshotIndex      = 1;
+        this.errorScreenshotIndex = 1;
+        this.pathCustomized       = false;
     }
 
-    static _getFileName (stepName) {
-        return `${stepName && stepName.replace(/\s|\\|\/|"|\*|\?|<|>|\|/g, '_') || 'page-load'}.png`;
+    static _correctFilePath (path) {
+        var correctedPath = path
```

Comment:
`sanitizedPath`

and do we need to perform split here? I thought `sanitize` can handle it